test/SwapFeeManagerTest.js

```diff
@@ -0,0 +1,91 @@
+const { expect } = require("chai");
+const { ethers } = require("hardhat");
+require('chai').use(require('chai-as-promised')).should();
+
+describe("SwapFeeManager", function () {
+    beforeEach(async function () {
+        accounts = await ethers.getSigners();
+
+        SwapFeeManager = await ethers.getContractFactory("SwapFeeManager");
+        swapFeeManager = await SwapFeeManager.deploy(
+            accounts[2].address, // dexFeeWallet
+            accounts[3].address  // burnFeeWallet
+        );
+        await swapFeeManager.waitForDeployment();
+
+        Token = await ethers.getContractFactory("Token");
+        token = await Token.deploy();
+        await token.waitForDeployment();
+
+        await token.transfer(accounts[1].address, ethers.parseEther("100"));
+    });
+
+    it("should correctly split and withdraw Ether fees", async function () {
+        // Send Ether to the SwapFeeManager contract
+        await accounts[1].sendTransaction({
+            to: swapFeeManager.target,
+            value: ethers.parseEther("1"),
+        });
+
+        const managerBalance = await ethers.provider.getBalance(swapFeeManager.target);
+        expect(managerBalance).to.equal(ethers.parseEther("1"));
+
+        const initialDexFeeBalance = await ethers.provider.getBalance(accounts[2].address);
+        const initialBurnFeeBalance = await ethers.provider.getBalance(accounts[3].address);
+
+        await swapFeeManager.connect(accounts[0]).splitAndWithdraw().should.be.fulfilled;
+
+        const finalDexFeeBalance = await ethers.provider.getBalance(accounts[2].address);
+        const finalBurnFeeBalance = await ethers.provider.getBalance(accounts[3].address);
+
+        expect(finalDexFeeBalance - initialDexFeeBalance).to.equal(ethers.parseEther("0.75"));
+        expect(finalBurnFeeBalance - initialBurnFeeBalance).to.equal(ethers.parseEther("0.25"));
```

Comment:
ummmm, why are we checking the balance difference and not just the new balance?

Reply:
Before each test we get a set of default hardhat accounts, which have non-zero eth balances 
https://github.com/KomodoPlatform/etomic-swap/blob/988190543cb1fa1efd2d23eac9776de1029f36db/test/SwapFeeManagerTest.js#L7
when you run tests, these accounts persist their state (including balance) across tests in the same testing session.

```
Initial ETH Dex Fee Wallet Balance: 10000000000000000000000
Initial ETH Burn Fee Wallet Balance: 10000000000000000000000
Final ETH Dex Fee Wallet Balance: 10000750000000000000000
Final ETH Burn Fee Wallet Balance: 10000250000000000000000
    ✔ should correctly split and withdraw Ether fees (61ms)
Initial ERC20 Dex Fee Wallet Balance: 0
Initial ERC20 Burn Fee Wallet Balance: 0
Initial ETH Dex Fee Wallet Balance: 10000750000000000000000
Initial ETH Burn Fee Wallet Balance: 10000250000000000000000
Final ERC20 Dex Fee Wallet Balance: 750000000000000000
Final ERC20 Burn Fee Wallet Balance: 250000000000000000
    ✔ should correctly split and withdraw ERC20 token fees (113ms)
```

As for Erc20 token, in log we see 0, as we dont transfer tokens manually to account as in these for example from EtomicSwapTakerV2.js
https://github.com/KomodoPlatform/etomic-swap/blob/988190543cb1fa1efd2d23eac9776de1029f36db/test/EtomicSwapTakerV2Test.js#L67

well, I can set balance to 0 with hardhat rpc https://github.com/KomodoPlatform/etomic-swap/pull/8/commits/4c0f115e2a8171127d9ca486a73ae6ab3b850241